src/tests/ftest/util/spdk.py

```diff
@@ -116,5 +116,33 @@ def nvme(hostlist, operation="cleanup"):
     #Check failure message in queue, and raise SpdkFailed if any thread Fails
     while not out_queue.empty():
         if out_queue.get() == "FAIL":
-            raise SpdkFailed("{} Failed".format(operation))
-    print("NVMe server {} Finished......".format(operation))
+            raise SpdkFailed("SPDK {} Failed".format(_oper[0]))
+    print("NVMe server {} Finished......".format(_oper[0]))
+
+def nvme_setup(hostlist):
+    """
+    nvme setup function called from Avocado test, This will start
+    thread for each server doing SPDK setup.
+    Args:
+        hostlist[list]: List of servers hostname.
+    return:
+        None
+    """
+    try:
+        nvme_main(hostlist, True)
+    except SpdkFailed:
+        raise
```

Comment:
Isn't this the same without the `try:`/`except:` handling?  I.e. all exceptions are raised to the caller.

Reply:
yes not needed extra block.